lighthouse-core/audits/preload-lcp-image.js

```diff
@@ -52,7 +52,7 @@ class PreloadLCPImageAudit extends Audit {
     // It's not a request loaded over the network, don't recommend it.
     if (URL.NON_NETWORK_PROTOCOLS.includes(request.protocol)) return false;
     // It's already discoverable from the main document, don't recommend it.
-    if (initiatorPath.length <= mainResourceDepth) return false;
+    if (initiatorPath.length <= mainResourceDepth + 1) return false;
```

Comment:
add a test for this case then? :)

Right now I'm not sure why we need `+ 1` if the `initiatorPath` does not include the node itself. Nodes discoverable from the main resource should have an initiator path length *equal to* the main document depth (because the initiator *was* the main document), no?

Maybe a comment on the `.slice` below or the `findLCPNode` method is warranted in either case to explain this :)

Reply:
Yeah, not sure why I felt like I had this wrong before. Gonna revert this change. I do, however, still need to make changes to also check if it is discoverable from the main resource but has a low priority.